_test/test.go

```diff
@@ -0,0 +1,98 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"time"
+
+	_ "github.com/joho/godotenv/autoload"
+	"github.com/libdns/libdns"
+	porkbun "github.comn/Niallfitzy1/libdns-porkbun"
+)
+
+func main() {
+
+	apikey := os.Getenv("PORKBUN_API_KEY")
+	secretapikey := os.Getenv("PORKBUN_SECRET_API_KEY")
+	zone := os.Getenv("ZONE")
+
+	if apikey == "" || secretapikey == "" || zone == "" {
+		fmt.Println("All variables must be set in '.env' file")
+		return
+	}
+
+	provider := porkbun.Provider{
+		APIKey:       apikey,
+		APISecretKey: secretapikey,
+	}
+
+	//Check Authorization
+	_, err := provider.CheckCredentials(context.TODO())
```

Comment:
If this is required, it won't meet our criteria of no extra setup or provisioning steps allowed (since the providers need to be interchangeable).

Reply:
Not required, it just checks credentials are valid which was handy when setting things up initially